### ISSUE TITLE
[PSyIR] Add support for Fortran CYCLE/C continue

### ISSUE THREAD

**Issue by arporter** [stfc/PSyclone#863]
The fparser2 fronted creates a CodeBlock when it encounters a CYCLE statement. The PSyIR needs extending to capture this concept and the frontend then needs to handle it correctly.

**Comment by sergisiso** [comment]
I agree we need the front-end handler, but maybe we can do without a PSyIR concept as it can be canonicalised to a simpler expression by adding the remainder of the loop inside a conditional:
From:
```
do i = 1, 10
   statement1
   if(condition) cycle
   statement2
   statement3
enddo
```
to:
```
do i = 1, 10
   statement1
   if (.not. condition) then
       statement2
       statement3
   endif
enddo
```
Would this work for your case?

**Comment by rupertford** [comment]
It think it would be too messy to implement this with nested conditions (hence the use of this concept in the first place I guess) so I think we would need the concept of cycle/continue in the general case. If it is needed then we might as well use it is the above example. Unless we ban nested conditions with cycle contained in them - but I don't like that.

**Comment by sergisiso** [comment]
Maybe you are right, I did try an example with multiple cycles in the if and else branches of a nested condition and it becomes complex very quickly. I am just worried that as more concepts there is, more edge cases there can be in the transformations.

Also worth considering that when inlining, 'kernel returns' may be converted to 'PSy loop cycles' and when fusing loops that have cycles this probably have to be converted to the conditional format anyway. (tbh I don't know if these examples are in support or against having the concept :))

**Comment by arporter** [comment]
@rupertford, this is (yet) another one that occurs at least once in the OCE part of NEMO. Probably lower a priority than the other ones I've flagged so far.
